libs/state/actions/src/lib/docs.spec.ts

```diff
@@ -0,0 +1,317 @@
+
+  private loginEffect = this.actions.onLogin(
+    (credentials$) =>
+      credentials$.pipe(
+        exhaustMap((credentials) => this.service.login(credentials))
+      ),
+    () => this.doc.defaultView.alert('successfully logged in')
+  );
+  constructor(
+    private service: AuthService,
+    @Inject(DOCUMENT) private doc: Document
+  ) {}
+}
+
+describe('handling side effects on event emission', () => {
+  let fixture: ComponentFixture<Login2Component>;
+  const documentMock = {
+    defaultView: {
+      alert: (v) => v,
+    },
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [Login2Component],
+      providers: [{ provide: AuthService, useClass: AuthServiceMock }],
+    })
+      .overrideComponent(Login2Component, {
+        set: {
+          providers: [
+            {
+              provide: DOCUMENT,
+              useValue: documentMock,
+            },
+          ],
+        },
+      })
+      .compileComponents();
+    fixture = TestBed.createComponent(Login2Component);
+    fixture.detectChanges();
+  });
+
+  it('should alert success after login', () => {
+    // arrange
+    const username = fixture.debugElement.query(By.css('input:first-child'));
+    const password = fixture.debugElement.query(
+      By.css('input[type="password"]')
+    );
+    const btn = fixture.debugElement.query(By.css('button'));
+    const alertSpy = jest.spyOn(documentMock.defaultView, 'alert');
+    username.nativeElement.value = 'user';
+    password.nativeElement.value = 'pwd';
+
+    // act
+    fixture.detectChanges();
+    btn.nativeElement.click();
+
+    // assert
+    expect(alertSpy).toHaveBeenCalled();
+  });
+});
+
+// unsubscribing from events programmatically
+
+describe('unsubscribing from events programmatically', () => {
+  let service: MovieService;
+  let resource: MovieResourceMock;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        MovieService,
+        { provide: MovieResource, useClass: MovieResourceMock },
+      ],
+    }).compileComponents();
+    service = TestBed.inject(MovieService);
+    resource = TestBed.inject(MovieResource) as any;
+  });
+
+  it('should fetching when disabled', () => {
+    // arrange
+    const movies = [{ id: '1' }];
+    service.disable();
+    // act
+    service.refresh();
+    resource.httpResponse.next(movies);
+    // assert
+    expect(service.movies()).toEqual([]);
+  });
+});
+
+// transform functions
+
+@Component({
+  template: `
+    <input name="name" #input (input)="ui.greet(input.value)" />
+    <div>{{ ui.greet$ | async }}</div>
+  `,
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: 'app-greet',
+  standalone: true,
+  imports: [AsyncPipe],
+})
+class GreetComponent {
+  ui = rxActions<{ greet: string }>(({ transforms }) =>
+    transforms({
+      // highlight-next-line
+      greet: (v) => `Hello ${v}`,
+    })
+  );
+}
+
+describe('transform functions', () => {
```

Comment:
What about testing also existing transform functions?

Reply:
Could you suggest a example pls